streaming/storage/subscription.go

```diff
@@ -0,0 +1,68 @@
+package storage
+
+import (
+	"context"
+	"errors"
+	"sync"
+)
+
+// A Subscription lets the client listen for changes in the given prefixed storage.
+// Subscriptions must be closed when not needed anymore.
+type Subscription struct {
+	cancel  context.CancelFunc
+	wg      sync.WaitGroup
+	changes <-chan struct{}
+	errors  <-chan error
+	closed  bool
+}
+
+type ChangesNotifierFunc func(ctx context.Context, changes chan<- struct{}) error
+
+func NewSubscription(ctx context.Context, subscriptionFunc ChangesNotifierFunc) *Subscription {
+	ctx, cancel := context.WithCancel(ctx)
+	changes := make(chan struct{})
+	errs := make(chan error, 1) // Buffered so there won't be a goroutine leak if nobody reads the error.
+	sub := &Subscription{
+		cancel:  cancel,
+		changes: changes,
+		errors:  errs,
+		wg:      sync.WaitGroup{},
+	}
+
+	sub.wg.Add(1)
+	go func() {
```

Comment:
I think that a constructor that starts a goroutine is quite peculiar. Maybe it's a Go construct that I haven't seen. You could add a short comment explaining why you do it in this specific place.

Reply:
Added comment